ui/src/app/views/settings/component.tsx

```diff
@@ -0,0 +1,154 @@
+      cluster: null,
+      resetClusterPending: false,
+      resetClusterModal: false,
+      clusterDeletionFailure: true
+    };
+  }
+
+  componentDidMount() {
+    this.pollCluster();
+  }
+
+  componentWillUnmount() {
+    if (this.pollClusterTimer) {
+      clearTimeout(this.pollClusterTimer);
+    }
+  }
+
+
+  render() {
+    return <React.Fragment>
+      <h2>Settings</h2>
+      <h3>Notebooks</h3>
+      <div style={styles.notebookSettings}>
+        <TooltipTrigger content={
+                          (!this.state.cluster) ?
+                            'Your notebook server is still being created' : undefined}
+                        side='right'>
+          <Button disabled={!this.state.cluster} onClick={this.openResetClusterModal.bind(this)}
+                  data-test-id='reset-notebook-button' type='secondary'>
+            Reset Notebook Server
+          </Button>
+        </TooltipTrigger>
+      </div>
+      {this.state.resetClusterModal &&
+      <Modal data-test-id='reset-notebook-modal'>
+        <ModalTitle>Reset Notebook Server?</ModalTitle>
+        <ModalBody>
+          <strong>Warning:</strong> Any unsaved changes to your notebooks may be lost
+          and your cluster will be offline for 5-10 minutes.
+          <br/><br/>
+          Resetting should not be necessary under normal conditions. Please help us to
+          improve this experience by using "Report a Bug" from the profile drop-down
+          to describe the reason for this reset.
+        </ModalBody>
+        <ModalFooter>
+          {this.state.clusterDeletionFailure ?
+            <div className='error'>Could not reset your notebook server.</div> : undefined}
+          <Button type='secondary'
+                  onClick={() => this.setState({resetClusterModal: false})}>Cancel</Button>
+          <Button disabled={this.state.resetClusterPending}
+                  onClick={() => this.resetCluster()}>Send</Button>
+        </ModalFooter>
+      </Modal>}
+    </React.Fragment>;
+  }
+
+  openResetClusterModal(): void {
+    this.setState({
+      resetClusterPending: false,
+      resetClusterModal: true,
+      clusterDeletionFailure: false
+    });
+  }
+
+  resetCluster(): void {
+    const args: FetchArgs = ClusterApiFetchParamCreator()
+      .deleteCluster(this.state.cluster.clusterNamespace, this.state.cluster.clusterName);
+    fetch(fullUrl(args.url), args.options)
+      .then(handleErrors)
+      .then(() => {
+        this.setState({cluster: null, resetClusterPending: false, resetClusterModal: false});
+        this.pollCluster();
+      })
+      .catch((error) => {
+        this.setState({resetClusterPending: false, clusterDeletionFailure: true});
+      });
+  }
+
+  private pollCluster(): void {
+    const repoll = () => {
+      this.pollClusterTimer = setTimeout(() => this.pollCluster(), 15000);
+    };
+    const args: FetchArgs = ClusterApiFetchParamCreator()
+      .listClusters();
+    console.log(args);
```

Comment:
debug line?  or do we want this written to the console?

Reply:
Removing